awswrangler/s3/_select.py

```diff
@@ -176,34 +256,40 @@ def select_query(
         raise exceptions.InvalidArgumentCombination(
             "'gzip' or 'bzip2' are only valid for input 'CSV' or 'JSON' objects."
         )
-    bucket, key = _utils.parse_path(path)
 
-    args: Dict[str, Any] = {
-        "Bucket": bucket,
-        "Key": key,
-        "Expression": sql,
-        "ExpressionType": "SQL",
-        "RequestProgress": {"Enabled": False},
-        "InputSerialization": {
-            input_serialization: input_serialization_params,
-            "CompressionType": compression.upper() if compression else "NONE",
-        },
-        "OutputSerialization": {
-            "JSON": {},
-        },
+    paths: List[str] = _path2list(
+        path=path,
+        boto3_session=boto3_session,
+        suffix=path_suffix,
+        ignore_suffix=_get_path_ignore_suffix(path_ignore_suffix=path_ignore_suffix),
+        ignore_empty=ignore_empty,
+        last_modified_begin=last_modified_begin,
+        last_modified_end=last_modified_end,
+        s3_additional_kwargs=s3_additional_kwargs,
+    )
+    if len(paths) < 1:
+        raise exceptions.NoFilesFound(f"No files Found on: {path}.")
+    _logger.debug("paths:\n%s", paths)
+
+    kwargs: Dict[str, Any] = {
+        "sql": sql,
+        "input_serialization": input_serialization,
+        "input_serialization_params": input_serialization_params,
+        "compression": compression,
+        "scan_range_chunk_size": scan_range_chunk_size,
+        "use_threads": use_threads,
+        "boto3_session": boto3_session,
+        "s3_additional_kwargs": s3_additional_kwargs,
     }
-    if s3_additional_kwargs:
-        args.update(s3_additional_kwargs)
-    _logger.debug("args:\n%s", pprint.pformat(args))
+    _logger.debug("kwargs:\n%s", pprint.pformat(kwargs))
 
-    if any(
-        [
-            compression,
-            input_serialization_params.get("AllowQuotedRecordDelimiter"),
-            input_serialization_params.get("Type") == "Document",
-        ]
-    ):  # Scan range is only supported for uncompressed CSV/JSON, CSV (without quoted delimiters)
-        # and JSON objects (in LINES mode only)
-        _logger.debug("Scan ranges are not supported given provided input.")
-        return pd.DataFrame(_select_object_content(args=args, boto3_session=boto3_session))
-    return _paginate_stream(args=args, path=path, use_threads=use_threads, boto3_session=boto3_session)
+    return concat_tables(
+        _utils.flatten_list(
+            *_read_tables_from_multiple_paths(
```

Comment:
`_read_tables_from_multiple_paths` should just return a list of tables, that does not need flattening or unpacking - not sure why these are needed here

Reply:
So you want to flatten/unpack the tables inside the `_read_tables_from_multiple_paths` method or are you asking why is flattening needed in the first place?

Reply:
According to `_read_tables_from_multiple_paths ` return type (and briefly looking at the source), it returns `List[Table]` - and I agree this is how it should be. 

But what you're doing with the result indicates that it returns a tuple that contains a list of lists of tables (a nested list of tables)?

Reply:
Correct, it would return a `List[List[Table]]`

Reply:
There's also a `*` unpacking operator so it must be returning a `Tuple[List[List[Table]]]`. Can't it just return a `List[Table]`?

Reply:
I have simplified the method to return `List[Table]`

Reply:
Cool, thanks